url2a908001cd_malictus_klang_tgz-pJ8-malictus_klang_test_PrimitiveTestJ8/src/malictus/klang/chunk/TEXTChunk.java

```diff
@@ -0,0 +1,130 @@
+			keywordData = new PrimitiveData(keyword, KlangConstants.PRIMITIVE_DATA_TEXT_CHUNK_KEYWORD, bytepos, null);
+		}
+		if (keepgoing && (raf.getFilePointer() < dataendpos) ) {
+			try {
+				bytepos = raf.getFilePointer();
+				text = new PlainString(raf, dataendpos, "ISO-8859-1", false);
+				textData = new PrimitiveData(text, KlangConstants.PRIMITIVE_DATA_TEXT_CHUNK_VALUE, bytepos, raf.getFilePointer());
+			} catch (Exception x) {
+				try {
+					text = new PlainString("", "ISO-8859-1", false);
+					textData = new PrimitiveData(text, KlangConstants.PRIMITIVE_DATA_TEXT_CHUNK_VALUE, bytepos, raf.getFilePointer());
+				} catch (Exception e) {} // shouldn't happen
+			}
+		} else {
+			try {
+				text = new PlainString("", "ISO-8859-1", false);
+				textData = new PrimitiveData(text, KlangConstants.PRIMITIVE_DATA_TEXT_CHUNK_VALUE, bytepos, raf.getFilePointer());
+			} catch (Exception e) {} // shouldn't happen
+		}
+		
+		/*
+		 * ADD PRIMITIVES TO VECTOR
+		 */
+		this.primitives.add(keywordData);
+		this.primitives.add(textData);
+	}
+	
+	/**
+	 * Overwrite of this method in Chunk class to write text data.
+	 * @param raff a pointer to the file (position is not important)
+	 * @throws IOException if any read/write error occurs
+	 */
+	public void reparseChunkPrimitives(RandomAccessFile raff) throws IOException {
+		if ((this.primitives == null) || (this.primitives.size() == 0)) {
+			//nothing to do!
+			return;
+		}
+		Primitive p1 = primitives.get(0).getPrimitive();
+		Primitive p2 = primitives.get(1).getPrimitive();
+		long oldsize = this.getDataEndPosition() - this.getDataStartPosition();
+		long pos = 0;
+		try {
+			long newsize = 0;
+			if (p1.valueExists()) {
+				newsize = p1.writeValueToString().getBytes(((StringPrim)p1).getTextEncoding()).length + 1;
+			}
+			if (p2.valueExists()) {
+				newsize = newsize + p2.writeValueToString().getBytes(((StringPrim)p2).getTextEncoding()).length;
+			}
+			if (this.getParentChunk() != null) {
+				if (this.getParentChunk().usesPadByte()) {
+					newsize = KlangUtil.adjustForPadByte(newsize);
+				}
+			}
+			long totnewsize = newsize + (this.getEndPosition() - this.getStartPosition()) - (this.getDataEndPosition() - this.getDataStartPosition());
+			if (newsize < oldsize) {
+				this.chunkIsAboutToChangeSize(totnewsize, raff);
+				KlangUtil.deleteFromFile(this.getParentFile().getFile(), this.getDataStartPosition(), this.getDataStartPosition() + (oldsize - newsize));
+			} else if (newsize > oldsize) {
+				this.chunkIsAboutToChangeSize(totnewsize, raff);
+				int amt = (int)(newsize - oldsize);
+				byte[] newbyte = new byte[amt];
+				KlangUtil.insertIntoFile(newbyte, this.getParentFile().getFile(), this.getDataStartPosition());
+			}
+			raff.seek(this.getDataStartPosition());
+			pos = raff.getFilePointer();
+			//write values back to the file
+			p1.writeValueToFile(raff);
+			if (p2 != null) {
+				p2.writeValueToFile(raff);
+			}			
+		} catch (BadValueException err) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 120.